src/core/renderers/canvas/utils/CanvasMaskManager.js

```diff
@@ -153,12 +208,63 @@ export default class CanvasMaskManager
     }
 
     /**
+     * Makes the black on the giben context transparent. This is necessary to make this canvas mask act as the
+     * webgl version with black masks.
+     * @param {Context} context - The context in which we want to convert black pixels into transparent.
+     */
+    _makeBlackTransparent(context)
+    {
+        const imgd = context.getImageData(0, 0, context.canvas.width, context.canvas.height);
+        const pix = imgd.data;
+
+        for (let i = 0, n = pix.length; i < n; i += 4)
+        {
+            if (pix[i] === 0 && pix[i + 1] === 0 && pix[i + 2] === 0)
+            {
+                pix[i + 3] = 0;
+            }
+        }
+
+        context.putImageData(imgd, 0, 0);
+    }
+
+    /**
      * Restores the current drawing context to the state it was before the mask was applied.
      *
      * @param {PIXI.CanvasRenderer} renderer - The renderer context to use.
      */
     popMask(renderer)
     {
+        this._freeCanvasOnNextFrame = this._freeCanvasOnNextFrame || [];
+        this._freeCanvasOnNextFrame.forEach((canvas) => this._pool.freeElement(canvas));
+
+        if (this._maskStates.length > 0)
+        {
+            const maskState = this._maskStates.pop();
+            const key = this._pool.getKey(renderer.context.canvas);
+            const maskCanvas = this._pool.popElementOrCreate(key, this._copyCanvas.bind(null, renderer));
+            const maskContext = this._setupContext(maskCanvas, renderer, maskState.maskData);
+
+            renderer.context = maskContext;
+
+            // Render the mask data to cull the content
+            maskState.maskData._renderCanvas(renderer);
+            // Black is transparent with the webgl mask, this forces this to be transparent too.
+            this._makeBlackTransparent(maskContext);
+            // Render the stuff to be masked into the mask context
+            maskContext.globalCompositeOperation = 'source-in';
+            maskContext.drawImage(maskState.maskableContext.canvas, -maskCanvas.width / 2, -maskCanvas.height / 2);
+            this._pool.freeElement(maskState.maskableContext.canvas);
+
+            // Restore the original context
+            renderer.context = maskState.originalContext;
+            // draw the masked content into the original context
+            renderer.context.drawImage(maskCanvas, -maskCanvas.width / 2, -maskCanvas.height / 2);
+            // freeing the maskCanvas here makes it create weird artifacts for some reason I haven't
+            // been able to figure out, if someone knows of a better solution, please fix this.
+            this._clearOnNextFrame.push(maskCanvas);
```

Comment:
I don't like this, but I haven't been able to figure out why this canvas is needed after this point (I don't think it should be necessary) and I couldn't think of any other point to free it.